pan/types.pan

```diff
@@ -829,6 +829,37 @@ type type_uppercase = string with {
 
 
 @documentation{
+Checks if the argument is a valid top level domain name (e.g. .example)
+}
+function is_top_level_domain = {
+    return(match(ARGV[0], '^(\.)+[a-zA-Z]{2,63}?$'));
+};
+
+
+type type_top_level_domain = string with {
+    is_top_level_domain(SELF);
+};
+
+
+function is_ipv4_netmask_pair = {
+    pair = split('\/', 1, ARGV[0]);
+    if (length(pair) == 2) {
+        ip = pair[0];
+        netmask = pair[1];
+        if (is_ip(ip) && (is_ip(netmask) || match(netmask, '^(\d+)$'))) {
```

Comment:
This bit conflates netmasks and prefix lengths, perhaps it is worth defining types for them?